common/src/main/java/us/ilite/common/lib/control/PIDController.java

```diff
@@ -0,0 +1,331 @@
+ * 
+ * Does all computation synchronously (i.e. the calculate() function must be called by the user from his own thread)
+ */
+public class PIDController {
+    private double m_P; // factor for "proportional" control
+    private double m_I; // factor for "integral" control
+    private double m_D; // factor for "derivative" control
+    private double m_F; // factor for feed forward gain
+    private double m_maximumOutput = 10.0; // |maximum output|
+    private double m_minimumOutput = -10.0; // |minimum output|
+    private double m_maximumInput = 0.0; // maximum input - limit setpoint to
+                                         // this
+    private double m_minimumInput = 0.0; // minimum input - limit setpoint to
+                                         // this
+    private boolean m_continuous = false; // do the endpoints wrap around? eg.
+                                          // Absolute encoder
+    private double m_prevError = 0.0; // the prior sensor input (used to compute
+                                      // velocity)
+    private double m_totalError = 0.0; // the sum of the errors for use in the
+                                       // integral calc
+    private double m_setpoint = 0.0;
+    private double m_error = 100.0;
+    private double m_result = 0.0;
+    private double m_last_input = Double.NaN;
+    private double m_deadband = 0.0; // If the absolute error is less than
+                                     // deadband
+                                     // then treat error for the proportional
+                                     // term as 0
+
+    public PIDController() {
+    }
+
+    /**
+     * Allocate a PID object with the given constants for P, I, D
+     *
+     * @param Kp
+     *            the proportional coefficient
+     * @param Ki
+     *            the integral coefficient
+     * @param Kd
+     *            the derivative coefficient
+     */
+    public PIDController(double Kp, double Ki, double Kd) {
+        m_P = Kp;
+        m_I = Ki;
+        m_D = Kd;
+        m_F = 0;
+    }
+
+    /**
+     * Allocate a PID object with the given constants for P, I, D
+     *
+     * @param Kp
+     *            the proportional coefficient
+     * @param Ki
+     *            the integral coefficient
+     * @param Kd
+     *            the derivative coefficient
+     * @param Kf
+     *            the feed forward gain coefficient
+     */
+    public PIDController(double Kp, double Ki, double Kd, double Kf) {
+        m_P = Kp;
+        m_I = Ki;
+        m_D = Kd;
+        m_F = Kf;
+    }
+
+    /**
+     * Read the input, calculate the output accordingly, and write to the output. This should be called at a constant
+     * rate by the user (ex. in a timed thread)
+     *
+     * @param input
+     *            the input
+     * @param dt
+     *            time passed since previous call to calculate
+     */
+    public double calculate(double input, double dt) {
+        if (dt < 1E-6)
```

Comment:
Add braces to ALL if(), for() and case() statements, even if they're only one line.